### PR TITLE
th-15: Websockets integration

### PR DESCRIPTION
**Pull request by DmytroVoronkov**
- Created websocket services in frontend and backend
- Created websocket handshake
image

### CODE REVIEW

File: frontend/src/libs/components/app/app.tsx [hunk 1]
```diff
@@ -8,19 +8,35 @@ import {
...
+  const [isWebSocketsConnected, setIsWebSocketsConnected] = useState(false);
   const { users, dataStatus } = useAppSelector(({ users }) => ({
     users: users.users,
     dataStatus: users.dataStatus,
   }));
 
   const isRoot = pathname === AppRoute.ROOT;
 
+  useEffect(() => {
+    const io = socket.getInstance();
```
Comment (Druxight) [inline frontend/src/libs/components/app/app.tsx]: Create a separate service responsible for socket encapsulation

**Review by Druxight — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/app/app.tsx [hunk 2]
```diff
@@ -8,19 +8,34 @@ import {
   useAppSelector,
   useEffect,
   useLocation,
+  useState,
 } from '~/libs/hooks/hooks.js';
+import { SocketService } from '~/libs/packages/socket/socket.service';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/app/app.tsx]:
```suggestion
import { SocketService } from '~/libs/packages/packages.js';
```

File: frontend/src/libs/packages/socket/libs/enums/enum.ts
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/socket/libs/enums/enum.ts:1]: file naming - enums.ts

File: frontend/src/libs/packages/socket/socket.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { type Socket } from 'socket.io-client';
+
+import { ClientSocketEvent } from '~/libs/packages/socket/libs/enums/enum.js';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/socket/socket.service.ts]: you are already in ~/libs/packages/socket folder

File: frontend/src/libs/packages/socket/socket.package.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { socket } from './socket.js';
+
+class SocketService {
+  private io: Socket;
+
+  public constructor() {
+    this.io = socket.getInstance();
+  }
+
+  public addListener(
```
Comment (Druxight) [inline frontend/src/libs/packages/socket/socket.package.ts:14]: why do we have 2 separate services when we can have only 1 ?

**Review by Druxight — CHANGES_REQUESTED** [review]

File: frontend/src/libs/packages/socket/libs/enums/enums.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export { ClientSocketEvent } from 'shared/build/index';
```
Comment (PavloSukhinin) [inline frontend/src/libs/packages/socket/libs/enums/enums.ts]: add file extension, please

File: shared/src/libs/packages/socket/libs/enums/socket-namespaces.enum.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const SocketNamespace = {
+  GLOBAL: '',
```
Comment (PavloSukhinin) [inline shared/src/libs/packages/socket/libs/enums/socket-namespaces.enum.ts]: Let's give it a name? I believe 'global' is enough

**Review by PavloSukhinin — DISMISSED** [review]

**Review by kalinkaaaa14 — APPROVED** [review]

**Review by PavloSukhinin — APPROVED** [review]

**Pull request merged**